lib/plugins/output-filter/kubernetes-enrichment.js

```diff
@@ -0,0 +1,212 @@
+  if (config.client && config.client.api) {
+    var spec = await config.client.api.v1.namespaces(namespace).pods(podName).get()
+    cb(null, spec.body)
+  } else {
+    cb(new Error('Kuberntes API not ready'))
+  }
+}
+
+function removeFields (key, data) {
+  if (podCache[key].stRemoveFields === false) {
+    return
+  }
+  var annotations = podCache[key].metadata.annotations
+  var removeFields = annotations['REMOVE_FIELDS'] || annotations['sematext.com/logs-remove-fields']
+  if (removeFields) {
+    var fieldNames = removeFields.split(',')
+    for (var i = 0; i < fieldNames.length; i++) {
+      delete data[fieldNames[i]]
+    }
+  } else {
+    podCache[key].stRemoveFields = false
+  }
+}
+
+function checkLogsEnabled (key, data, context) {
+  if (podCache[key].stLogEnabled === false) {
+    data.stLogEnabled = false
+    return
+  }
+  var annotations = podCache[key].metadata.annotations
+  if (annotations) {
+    var logsEnabled = annotations['sematext.com/logs-enabled']
+    if (FALSE_REGEX.test(logsEnabled)) {
+      data.stLogEnabled = false
+      podCache[key].stLogEnabled = false
+      if (process.env.DEBUG) {
+        console.error('stLogEnabled = false', key)
+      }
+    } else {
+      podCache[key].stLogEnabled = true
+    }
+  }
+}
+
+function addLogsIndex (key, data) {
+  if (podCache[key].stLogsTokenSet === false) {
+    return
+  }
+  // get ST Logs Token from pod annotation
+  var index = podCache[key].metadata.annotations['sematext.com/logs-token']
+  // get comma separated list of fields from pod annotation
+  if (index) {
+    data._index = index
+  } else {
+    podCache[key].stLogsTokenSet = false
+  }
+}
+/*
+
+function updatePodUid (key, data) {
+  if (data.kubernetes && podCache[key]) {
+    data.kubernetes.pod.uid = podCache[key].metadata.uid
+  } else {
+    podCache[key].stLogsTokenSet = false
+  }
+}
+
+*/
+
+function replaceDockerImageName (key, data) {
+  if (data.kubernetes && data.container && podCache[key].stImageCache) {
+    var containerName = data.kubernetes.pod.container.name
+    var imageInfo = podCache[key].stImageCache[data.kubernetes.pod.name + containerName]
+    if (!imageInfo) {
+      return
+    }
+    if (data.container.image.name === 'sha256') {
+      data.container.image.digest = 'sha256:' + data.container.image.tag
+    }
+    data.container.image.name = imageInfo.name
+    data.container.image.tag = imageInfo.tag
+    // data.container.image.registry = imageInfo.version
+    // data.container.image.name = imageInfo.plainImageName
+  } else {
+    podCache[key].stLogsTokenSet = false
+  }
+}
+
+function processAnnotations (data, context) {
+  var key = data.kubernetes.namespace + '/' + data.kubernetes.pod.name
+  if (podCache[key] && podCache[key].metadata) {
+    checkLogsEnabled(key, data, context)
+    if (data.stLogEnabled === false) {
+      // logs will be droped, so no need for other processing
+      return
+    }
+    // updatePodUid (key, data)
```

Comment:
Delete commented code?

Reply:
There is potentially a problem in different UID formats, so the function was experimental to fix it. I will remove it, and potentially add later again if it turns out that `sematext/agent` use the UID from k8s API as well. We have a task to sync sematext/agent tags/values and Logagent tags/values.